src/Bicep.Core.UnitTests/TypeSystem/UnionTypeTests.cs

```diff
@@ -105,11 +105,11 @@ public void SingletonUnionCreationShouldProduceSingletonType()
         }
 
         [TestMethod]
-        public void UnionsOfStringsAndStringLiteralTypesShouldProduceStringType()
+        public void UnionsOfStringsAndStringLiteralTypesShouldNotDropLiterals()
         {
-            TypeHelper.CreateTypeUnion(LanguageConstants.String, new StringLiteralType("hello"), new StringLiteralType("there")).Should().BeSameAs(LanguageConstants.String);
+            TypeHelper.CreateTypeUnion(LanguageConstants.String, new StringLiteralType("hello"), new StringLiteralType("there")).Name.Should().Be("'hello' | 'there' | string");
 
-            TypeHelper.CreateTypeUnion(LanguageConstants.String, new StringLiteralType("hello"), LanguageConstants.Bool, new StringLiteralType("there")).Name.Should().Be("bool | string");
+            TypeHelper.CreateTypeUnion(LanguageConstants.String, new StringLiteralType("hello"), LanguageConstants.Bool, new StringLiteralType("there")).Name.Should().Be("'hello' | 'there' | bool | string");
```

Comment:
>CreateTypeUnion [](http://example.com/codeflow?start=23&length=15)

Can you also add a case to assert that `'hello' | 'there' | string | string` becomes `'hello' | 'there' | string`?